### PR TITLE
use single cache

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-1122

### Change description ###
current version works with cachewriter  but  using single cache might be easier .
userdetails cache removed, change the cached object and return to requester.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClient.java [hunk 1]
```diff
@@ -24,68 +21,45 @@
...
-        log.info("Remove access token from cache for jurisdiction: {} ", jurisdiction);
-        accessTokenCache.invalidate(jurisdiction);
+        log.info("Remove idam credential from cache for jurisdiction: {} ", jurisdiction);
+        idamCache.invalidate(jurisdiction);
     }
 
-    private CachedIdamToken retrieveToken(String jurisdiction) {
-        log.info("Retrieve access token for jurisdiction: {} from IDAM", jurisdiction);
+    private CachedIdamCredential retrieveIdamInfo(String jurisdiction) {
+        log.info("Retrieving access token for jurisdiction: {} from IDAM", jurisdiction);
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClient.java:53]: I like `retrieving...`, as it leaves no doubt that we're only about to do it. Maybe we could also apply it in the other logs, where we currently have `get` and `remove`?

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java [hunk 2]
```diff
@@ -92,21 +95,27 @@ public void should_cache_by_jurisdiction() {
...
-            idamCachedClient.getAccessToken(jurisdiction2);
+        CachedIdamCredential cachedIdamCredential1 =
+            idamCachedClient.getIdamCredentials(jurisdiction1);
+        CachedIdamCredential cachedIdamCredential2 =
+            idamCachedClient.getIdamCredentials(jurisdiction2);
 
-        assertThat(JWT).isEqualTo(token1);
-        assertThat(JWT2).isEqualTo(token2);
+        assertThat(cachedIdamCredential1.accessToken).isEqualTo(JWT);
+        assertThat(cachedIdamCredential1.userDetails).isEqualTo(USER_DETAILS);
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java:112]: Worth replacing with `.isEqualTo(expectedUserDetails1`. This is not a part of this PR, though.

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java [hunk 3]
```diff
@@ -163,97 +175,63 @@ void should_throw_exception_when_token_decoding_fails() {
...
-            idamCachedClientQuickExpiry.getAccessToken(jurisdiction);
+        given(idamApi.authenticateUser(USERNAME, PASSWORD)).willReturn(JWT, JWT2);
+        given(idamApi.getUserDetails(JWT)).willReturn(USER_DETAILS);
+        UserDetails expectedUserDetails2 =  new UserDetails("12","q@a.com","","",null);
+        given(idamApi.getUserDetails(JWT2)).willReturn(expectedUserDetails2);
+
+        CachedIdamCredential cachedIdamCredential1 =
+            idamCachedClientQuickExpiry.getIdamCredentials(jurisdiction);
 
         TimeUnit.SECONDS.sleep(3);
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java:194]: Can we make it expire even quicker?

Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java:194]: expiry  is 2 second. I want to be sure expired as it is configured.

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java (hunk 2 above)
Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java]: changed

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClient.java (hunk 1 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClient.java:53]: added

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java (hunk 3 above)
Comment (nitinprabhu) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java:196]: worth adding a comment here why this sleep is added

Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/cache/IdamCachedClientTest.java:197]: comment added

**Pull request merged**